adapter/test/pubsub/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+#------------------------------------------------------------------------------
+# PubSub Adapter tests
+#------------------------------------------------------------------------------
+add_executable(pubsub_metadata_test ${CMAKE_CURRENT_SOURCE_DIR}/pubsub_metadata_test.cc)
+add_test(NAME "pubsub_metadata_test" COMMAND "${CMAKE_BINARY_DIR}/bin/pubsub_metadata_test")
+target_link_libraries(pubsub_metadata_test ${LIBRT} hermes MPI::MPI_CXX
+        $<$<BOOL:${HERMES_RPC_THALLIUM}>:thallium> hermes_pubsub)
+add_dependencies(pubsub_metadata_test hermes_pubsub)
+
+set(PUBSUB_TESTS pubsub_connection_test pubsub_topic_test
+        pubsub_end_to_end_test pubsub_end_to_end_test_sync)
+set(EXTENDED_PUBSUB_TESTS ${PUBSUB_TESTS} pubsub_metadata_test)
+foreach(program ${PUBSUB_TESTS})
+    add_executable(${program} ${CMAKE_CURRENT_SOURCE_DIR}/${program}.cc)
+    add_dependencies(${program} hermes_pubsub)
+    target_link_libraries(${program} ${LIBRT} hermes MPI::MPI_CXX
+            $<$<BOOL:${HERMES_RPC_THALLIUM}>:thallium> hermes_pubsub)
+    target_compile_definitions(${program}
+            PRIVATE $<$<BOOL:${HERMES_RPC_THALLIUM}>:HERMES_RPC_THALLIUM>)
+    mpi_daemon(${program} 1 "" "" 1)
+endforeach()
+
+foreach(program ${EXTENDED_PUBSUB_TESTS})
+    target_include_directories(${program} PUBLIC ${HERMES_ADAPTER_DIR})
+    target_include_directories(${program} PUBLIC ${HERMES_ADAPTER_TEST_DIR})
+endforeach()
+
+if(HERMES_INSTALL_TESTS)
+    set(PUBSUB_TESTS pubsub_connection_test pubsub_metadata_test pubsub_negative_connection_test pubsub_topic_test
+            pubsub_end_to_end_test pubsub_end_to_end_test_sync)
```

Comment:
Can you use `EXTENDED_PUBSUB_TESTS` here? Also, I don't see a test called `pubsub_negative_connection_test`.